### PR TITLE
Bug 1796926: baremetal: don't set default for provisioning interface

### PR DESCRIPTION
**Pull request by stbenjam**
The provisioning interface is the interface on a baremetal host
connected to the provisioning network. Setting a default here doesn't
make a lot of sense, as it's highly dependent on the user's hardware and
environment.

### CODE REVIEW

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1452/

File: pkg/types/baremetal/validation/platform.go [hunk 1]
```diff
@@ -105,7 +105,7 @@ func ValidatePlatform(p *baremetal.Platform, n *types.Networking, fldPath *field
 
 	// Make sure the provisioning interface is set.  Very little we can do to validate this  as it's not on this machine.
 	if p.ProvisioningNetworkInterface == "" {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("provisioningNetworkInterface"), p.ProvisioningNetworkInterface, "No baremetal provisioning network interface is configured."))
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("provisioningNetworkInterface"), p.ProvisioningNetworkInterface, "no provisioning network interface is configured."))
```
Comment (hardys) [inline pkg/types/baremetal/validation/platform.go]: I wonder if we should make the error a bit more helpful, e.g say something like "no ProvisioningNetworkInterface is configured, this should be set to the interface name connected to the provisioning network on the deployed cluster nodes" ?

**Comment by hardys** [comment]
lgtm, small nit re the validation message for your consideration.

File: pkg/types/baremetal/validation/platform.go (hunk 1 above)
Comment (stbenjam) [reply pkg/types/baremetal/validation/platform.go]: Updated, good suggestion thanks

**Comment by imain** [comment]
Yeah good plan guys.  I was thinking this myself.

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1457/

**Comment by hardys** [comment]
Something went wrong with the installer build in CI, I don't see exactly why in the logs and it builds for me locally so trying a retest

**Comment by yuvalk** [comment]
maybe also add an example to:
https://github.com/openshift/installer/blob/master/docs/user/metal/install_ipi.md
?

**Comment by stbenjam** [comment]
> maybe also add an example to:
> [/docs/user/metal/install_ipi.md@`master`](https://github.com/openshift/installer/blob/master/docs/user/metal/install_ipi.md) [](/openshift/installer/blob/HEAD@{2020-01-27T11:04:16Z}/docs/user/metal/install_ipi.md)
> ?

Thanks, good suggestion as this is now a required parameter.

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1461/

**Comment by stbenjam** [comment]
@hardy This needs a /approve to land, I filed a BZ for it.

**Pull request merged**